Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v9.md

```diff
@@ -0,0 +1,162 @@
+
+```csharp
+[Umbraco.Forms.Core.Attributes.Setting("Log Header",
+        Description = "Log item header",
+        View = "TextField")]
+public string LogHeader { get; set; }
+```
+
+The Umbraco.Forms.Core.Attributes.Setting registers the property in Umbraco Forms and there will automatically be UI and storage generated for it. In the attribute, a name, description and the view to be rendered is defined.
+
+With the attribute in place, the property value is set every time the class is instantiated by Umbraco Forms. This means you can use the property in your code like this:
+
+```csharp
+[Umbraco.Forms.Core.Attributes.Setting("Document ID",
+        Description = "Node the log entry belongs to",
+        View = "Pickers.Content")]
+public string Document { get; set; }
+
+public override WorkflowExecutionStatus Execute(WorkflowExecutionContext context) {
+    _logger.LogInformation("Record submitted from: {IP}", context.Record.IP);
+    return WorkflowExecutionStatus.Completed;
+}
+```
+
+For all types that use the provider model, settings work this way. By adding the Setting attribute Forms automatically registers the property in the UI and sets the value when the class is instantiated.
+
+## Validate type settings with ValidateSettings()
+
+The `ValidateSettings()` method which can be found on all types supporting dynamic settings, is used for making sure the data entered by the user is valid and works with the type.
+
+```csharp
+public override List<Exception> ValidateSettings() {
+    List<Exception> exceptions = new List<Exception>();
+    int docId = 0;
+    if (!int.TryParse(Document, out docId))
+        exceptions.Add(new Exception("Document is not a valid integer"));
+    return exceptions;
+}
+```
+
+## Registering the class with Umbraco and Forms
+
+To register the type, ensure your web application project has a reference to the class library - either via a project or NuGet reference - and add the following code into the startup pipeline.  In this example, the registration is implemented as an extension method to `IUmbracoBuilder` and should be called from `Startup.cs`:
+
+```csharp
+public static IUmbracoBuilder AddUmbracoFormsCoreProviders(this IUmbracoBuilder builder)
+{
+    builder.WithCollectionBuilder<WorkflowCollectionBuilder>()
+        .Add<LogWorkflow>();
+}
+```
+
+Also look in the reference chapter for complete class implementations of workflows, fields and export types.
+
+## Overriding default providers in Umbraco Forms
+
+It is possible to override and inherit the original provider, be it a Field Type or Workflow etc. The only requirement when inheriting a fieldtype that you wish to override is to ensure you do not override/change the Id set for the provider, and make sure your class is public.
+
+Here is an example of overriding the Textarea field aka Long Answer that is taken from Per's CodeGarden 17 talk, which has been updated for Forms 9.
```

Comment:
Shortened this sentence a bit, as the info about Per's Codegarden talk seemed a bit out of place!

```suggestion
Here is an example of overriding the Textarea field aka Long Answer.
```